src/midcom/datamanager/storage/parameter.php

```diff
@@ -29,6 +50,106 @@ public function save()
             && $this->load() === null) {
             return true;
         }
+
+        if (!empty($this->config['type_config']['multiple_storagemode'])) {
+            $this->value = $this->convert_multiple_to_storage();
```

Comment:
not entirely sure if this might be a problem, but if `save()` gets called twice for some reason, then `$this->value` would already be in the converted form during the second call. I wonder if it wouldn't be safer to use a local variable for the converted value